CMakeLists.txt

```diff
@@ -8,6 +8,7 @@ include(cmake/HelperBoost.cmake)
 include_directories(${Boost_INCLUDE_DIRS})
 
 find_package(OpenMP)
+find_package(Eigen3 3.3 REQUIRED)
```

Comment:
It'll be necessary to think twice before adding `Eigen` as a dependency.

Reply:
I agree. It seems to me it is basically required everywhere because it is required by the new edge collapser which is required almost everywhere.